### ISSUE TITLE
Java packages support

### ISSUE THREAD

**Issue by novakboskov** [jaypei/emacs-neotree#84]
Is there a method to visualize Java packages like this:
|-- org.something.controller
    |-- source.java
    |-- anotherSource.java
not like:
|-- org
    |-- something
        |-- controller
            |-- source.java
            |-- anotherSource.java
?

**Comment by jaypei** [comment]
Hi novakboskov,

NeoTree dose not support display java package.
I do not use JAVA, so I do not known how to recognize a JAVA project directory.
But I think NeoTree should provide a way to customize the tree content.

Jaypei

**Comment by vkazanov** [comment]
@jaypei So what do you think about abstracting away the tree content? Are you working on it or..?

I need a tool to display module/package structure of a project. Neotree looks cute, but directories/files are deeply interwinded with the code. What do you think about the idea of abstracting away from the concrete tree (files/directories)? I'd spend a few evenings on a prototype.

**Comment by jaypei** [comment]
Hi vkazanov,

So far I did not do this.

It should not be hard. Define a tree view layer, and the APIs can use prefix `neo-tree--`. The view can bound to a buffer (like  *NeoTree*). Plugins only focus on the tree API and event hooks.

This is what ECB does.

Jaypei

**Comment by vkazanov** [comment]
Yes, this is exactly what I have in mind: isolate a tree API, switch from using --current-path to using something like --current-node, etc. Should not be a problem.

**Comment by howdoicomputer** [comment]
This would be pretty cool.
